app/Filament/Pages/ArtificialIntelligence.php

```diff
@@ -0,0 +1,323 @@
+use Illuminate\Validation\Rules\Password;
+
+/**
+ * @property Form $form
+ */
+class ArtificialIntelligence extends Page
+{
+    use InteractsWithFormActions;
+
+    protected static string $view = 'filament.pages.profile-save';
+
+    protected static ?string $cluster = ProfileSettings::class;
+
+    protected static ?string $slug = 'artificial-intelligence';
+
+    protected static ?string $title = 'Artifical Intelligence';
+
+    protected static ?int $navigationSort = 40;
+
+    protected static bool $shouldRegisterNavigation = true;
+
+    /** @var array<string, mixed> $data */
+    public ?array $data = [];
+
+    public function form(Form $form): Form
+    {
+        return $form
+            ->schema([
+                Section::make('Artificial Intelligence')
+                    ->description('Select options for how you work with AI.')
+                    ->schema([
+                        Select::make('is_submit_ai_chat_on_enter_enabled')
+                            ->label('Enter Key')
+                            ->selectablePlaceholder(false)
+                            ->hint('Decide below if you would prefer the enter key to create a new line or submit the prompt you typed in the AI chat interface.')
+                            ->options([
+                                false => 'New Line',
+                                true => 'Enter',
+                            ]),
+                    ])
+                    ->visible(SubmitAiChatOnEnterFlag::active()),
+            ]);
+    }
+
+    public function mount(): void
+    {
+        $this->fillForm();
+    }
+
+    public function getUser(): Authenticatable|Model
+    {
+        $user = Filament::auth()->user();
+
+        if (! $user instanceof Model) {
+            throw new Exception('The authenticated user object must be an Eloquent model to allow the profile page to update it.');
+        }
+
+        return $user;
+    }
+
+    public function save(): void
+    {
+        try {
+            $this->callHook('beforeValidate');
+
+            $data = $this->form->getState();
+
+            $this->callHook('afterValidate');
+
+            $data = $this->mutateFormDataBeforeSave($data);
+
+            $this->callHook('beforeSave');
+
+            $this->handleRecordUpdate($this->getUser(), $data);
+
+            $this->callHook('afterSave');
+        } catch (Halt $exception) {
+            return;
+        }
+
+        if (request()->hasSession() && array_key_exists('password', $data)) {
+            request()->session()->put([
+                'password_hash_' . Filament::getAuthGuard() => $data['password'],
+            ]);
+        }
+
+        $this->data['password'] = null;
+        $this->data['passwordConfirmation'] = null;
+
+        $this->getSavedNotification()?->send();
+
+        $this->dispatch('refresh-branding-bar');
+
+        if ($redirectUrl = $this->getRedirectUrl()) {
+            $this->redirect($redirectUrl);
+        }
+    }
```

Comment:
I am noticing that these, and many other methods are exactly the same between all of the pages.

It would be best if we created something like an abstract base `ProfilePage` class or something like that that contained all of the shared method that are exactly the same and have all of these new pages extend that.

What do you think?